### PR TITLE
[VIVO-1900] i18n: added empty check for getCountry, fixing bug with spanish label (es)

### PR DESCRIPTION
**Pull request by dofeldsc**
**[JIRA Issue](https://jira.lyrasis.org/browse/VIVO-1900)**

[2nd PR](https://github.com/vivo-project/VIVO/pull/182)

# What does this pull request do?
This PR adds an check for empty Country language tags. Most tags are like this "en-US" or "de-DE" but the spanish one is just "es". The language tag consists of two parts and if the second part (Country) is empty the spanish label looked like this "es-" and caused an error.

I have corrected not only the part that causes this error but all the ones I have found which could also lead to this or similar errors.

# What's new?
Check for empty country language tag.

# How should this be tested?
With a fresh VIVO instanz and the given sample data (https://wiki.lyrasis.org/display/VIVODOC19x/Sample+Data) i tried to add an spanish label to an article. Go to the article, edit label with language set to spanish ( you do not need to edit it) and click "Editar etiqueta" (Edit Label). After that an error  occurred. If this does not happen and the label is set, it should be fine.

### CODE REVIEW

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/json/GetRandomSearchIndividualsByVClass.java [hunk 1]
```diff
@@ -75,7 +75,10 @@ private String renderShortView(String individualUri, String vclassName) {
 		modelMap.put("individual",
 				IndividualTemplateModelBuilder.build(individual, vreq));
 		modelMap.put("vclass", vclassName);
-		String langCtx = vreq.getLocale().getLanguage() + "-"+vreq.getLocale().getCountry();  //UQAM-Linguistic-Management build the linguistic context
+		String langCtx = vreq.getLocale().getLanguage();  //UQAM-Linguistic-Management build the linguistic context
+        if (!vreq.getLocale().getCountry().isEmpty()) {
```
Comment (awoods) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/json/GetRandomSearchIndividualsByVClass.java:79]: The formatting in the VIVO codebase is... inconsistent. However, it is helpful not to exacerbate the chaos. Would you mind changing the indentation of the lines you added to either use spaces or tabs in order to match the lines around it?

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/MultiValueEditSubmission.java [hunk 2]
```diff
@@ -303,7 +303,10 @@ public void addLiteralToForm(EditConfigurationVTwo editConfig, FieldVTwo field,
 					String rangeLang = field.getRangeLang();  //UQAM  Default value
 					try {
 						if (_vreq != null ) {
-							rangeLang	= _vreq.getLocale().getLanguage() + "-"+_vreq.getLocale().getCountry();
+                            rangeLang	= _vreq.getLocale().getLanguage();
```
Comment (awoods) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/MultiValueEditSubmission.java:306]: The formatting in the VIVO codebase is... inconsistent. However, it is helpful not to exacerbate the chaos. Would you mind changing the indentation of the lines you added to either use spaces or tabs in order to match the lines around it?

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/services/shortview/FakeApplicationOntologyService.java [hunk 3]
```diff
@@ -494,7 +494,10 @@ public FakeVivoPeopleDataGetter(VitroRequest vreq, String individualUri) {
 			this.individualUri = individualUri;
 			this.vreq = vreq;
 			this.ctx = vreq.getSession().getServletContext();
-			this.langCtx =  vreq.getLocale().getLanguage() + "-"+vreq.getLocale().getCountry(); // UQAM-Optimization add the linguistic context
+			this.langCtx =  vreq.getLocale().getLanguage(); // UQAM-Optimization add the linguistic context
+            if (!vreq.getLocale().getCountry().isEmpty()) {
```
Comment (awoods) [inline api/src/main/java/edu/cornell/mannlib/vitro/webapp/services/shortview/FakeApplicationOntologyService.java:498]: The formatting in the VIVO codebase is... inconsistent. However, it is helpful not to exacerbate the chaos. Would you mind changing the indentation of the lines you added to either use spaces or tabs in order to match the lines around it?

**Comment by dofeldsc** [comment]
I did the changes and additionally I have adapted a try - catch block so it is consistent in the function. Most of the time i love my IDE but sometimes it makes it harder...I hope it fits like this.

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/services/shortview/FakeApplicationOntologyService.java (hunk 3 above)
Comment (dofeldsc) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/services/shortview/FakeApplicationOntologyService.java]: i changed it.

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/MultiValueEditSubmission.java (hunk 2 above)
Comment (dofeldsc) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/MultiValueEditSubmission.java]: understandable, i changed it.

File: api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/json/GetRandomSearchIndividualsByVClass.java (hunk 1 above)
Comment (dofeldsc) [reply api/src/main/java/edu/cornell/mannlib/vitro/webapp/controller/json/GetRandomSearchIndividualsByVClass.java]: understandable, i changed it.

**Pull request merged**